crates/bevy_asset/src/io/embedded/mod.rs

```diff
@@ -299,6 +299,73 @@ mod tests {
         assert_eq!(asset_path, Path::new("my_crate/foo/the/asset.png"));
     }
 
+    // A blank src_path removes the embedded's file path altogether.
+    #[test]
+    fn embedded_asset_path_from_local_crate_blank_src_path_questionable() {
+        let asset_path = _embedded_asset_path(
+            "my_crate",
+            "".as_ref(),
+            "src/foo/some/deep/path/plugin.rs".as_ref(),
+            "the/asset.png".as_ref(),
+        );
+        assert_eq!(asset_path, Path::new("my_crate/the/asset.png"));
+    }
+
+    #[should_panic(expected = "Failed to find src_prefix \"src\" in \"\"")]
+    #[test]
+    fn embedded_asset_path_from_local_crate_blank_file_path() {
+        let asset_path = _embedded_asset_path(
+            "my_crate",
+            "src".as_ref(),
+            "".as_ref(),
+            "the/asset.png".as_ref(),
+        );
+        assert_eq!(asset_path, Path::new("my_crate/foo/the/asset.png"));
+    }
+
+    // Can have extraneous slashses at the end of source_path.
+    #[test]
+    fn embedded_asset_path_from_local_crate_extraneous_slalsh() {
```

Comment:
This is another condition that should be handled by `Path`. I'd drop this test.

Reply:
Deleted.